services/scraper/scraper.go

```diff
@@ -22,186 +23,126 @@ var selectors = map[string]string {
 
-	startScraping(queue)
+	return service.startScraping(url)
 }
 
-func setupQueue(user *models.User, keywords []string) *queue.Queue {
-	searchQueue, err := queue.New(2, &queue.InMemoryQueueStorage{MaxSize: 10000})
-	if err != nil {
-		logs.Error("Failed to create a queue:", err.Error())
-	}
-
-	for _, k := range keywords {
-		scrapingResults[k] = createResult(user, k)
+func (service *Scraper) startScraping(url string) error {
+	collector := colly.NewCollector()
 
-		escapedKeyword := url.QueryEscape(k)
-		err = searchQueue.AddURL(fmt.Sprintf(GOOGLE_SEARCH_URL, escapedKeyword))
-		if err != nil {
-			logs.Info("Failed to add url to queue:", err.Error())
-		}
-	}
+	collector.OnRequest(service.requestHandler)
+	collector.OnResponse(service.responseHandler)
+	collector.OnError(service.errorHandler)
 
-	return searchQueue
-}
-
-func createResult(user *models.User, keyword string) int64  {
-	result := &models.Result{
-		User: user,
-		Keyword: keyword,
-	}
-	resultID, err := models.CreateResult(result)
-	if err != nil {
-		logs.Error("Failed to create result:", err.Error())
-	}
-
-	return resultID
-}
-
-func startScraping(queue *queue.Queue)  {
-	async := helpers.GetAppRunMode() != "test"
-	collector := colly.NewCollector(colly.Async(async))
-
-	collector.OnRequest(requestHandler)
-	collector.OnResponse(responseHandler)
-	collector.OnError(errorHandler)
-
-	collector.OnHTML(selectors["wholePage"], wholePageCollector)
+	collector.OnHTML(selectors["wholePage"], service.wholePageCollector)
 
 	collector.OnHTML(selectors["nonAds"], func(e *colly.HTMLElement) {
-		addNonAdLinkToResult(e)
+		service.addNonAdLinkToResult(e)
 	})
 
 	collector.OnHTML(selectors["topImageAds"], func(e *colly.HTMLElement) {
-		addAdLinkToResult(models.AdLinkTypeImage, models.AdLinkPositionTop, e)
+		service.addAdLinkToResult(models.AdLinkTypeImage, models.AdLinkPositionTop, e)
 	})
 
 	collector.OnHTML(selectors["topLinkAds"], func(e *colly.HTMLElement) {
-		addAdLinkToResult(models.AdLinkTypeLink, models.AdLinkPositionTop, e)
+		service.addAdLinkToResult(models.AdLinkTypeLink, models.AdLinkPositionTop, e)
 	})
 
 	collector.OnHTML(selectors["sideImageAds"], func(e *colly.HTMLElement) {
-		addAdLinkToResult(models.AdLinkTypeImage, models.AdLinkPositionSide, e)
+		service.addAdLinkToResult(models.AdLinkTypeImage, models.AdLinkPositionSide, e)
 	})
 
 	collector.OnHTML(selectors["bottomAds"], func(e *colly.HTMLElement) {
-		addAdLinkToResult(models.AdLinkTypeLink, models.AdLinkPositionBottom, e)
+		service.addAdLinkToResult(models.AdLinkTypeLink, models.AdLinkPositionBottom, e)
 	})
 
-	collector.OnScraped(finishScrapingHandler)
+	collector.OnScraped(service.finishScrapingHandler)
 
-	err := queue.Run(collector)
+	err := collector.Visit(url)
 	if err != nil {
-		logs.Info("Failed to run the queue:", err.Error())
+		return err
 	}
+
+	return nil
 }
 
-func requestHandler(request *colly.Request) {
+func (service *Scraper) requestHandler(request *colly.Request) {
```

Comment:
Receiver name should be one or two letter 
```suggestion
func (s *Scraper) requestHandler(request *colly.Request) {
```

Reply:
fixed in [538f129](https://github.com/carryall/go-google-scraper-challenge/pull/71/commits/538f12914d771bf610ef2f9a64ef5b41254eb52c)